tests/util/util_head_object.sh

```diff
@@ -134,10 +134,45 @@ check_checksum_rest() {
     log 2 "expected response code '200', was '$result'"
     return 1
   fi
-  head_checksum=$(grep "x-amz-checksum-sha256" "$TEST_FILE_FOLDER/result.txt" | awk '{print $2}' | sed 's/\r$//')
-  file_checksum="$(sha256sum "$3" | awk '{print $1}' | xxd -r -p | base64)"
-  if [ "$file_checksum" != "$head_checksum" ]; then
+  head_checksum=$(grep -i "$4" "$TEST_FILE_FOLDER/result.txt" | awk '{print $2}' | sed 's/\r$//')
+  if [ "$3" != "$head_checksum" ]; then
     log 2 "'checksum mismatch (head '$head_checksum', local '$file_checksum')"
     return 1
   fi
 }
+
+check_checksum_rest_sha256() {
+  if [ $# -ne 3 ]; then
+    log 2 "'check_checksum_rest_sha256' requires bucket, file, local file"
+    return 1
+  fi
+  file_checksum="$(sha256sum "$3" | awk '{print $1}' | xxd -r -p | base64)"
+  if ! check_checksum_rest "$1" "$2" "$file_checksum" "x-amz-checksum-sha256"; then
+    log 2 "error checking checksum"
+    return 1
+  fi
+  return 0
+}
+
+check_checksum_rest_crc32() {
+  if [ $# -ne 3 ]; then
+    log 2 "'check_checksum_rest_crc32' requires bucket, file, local file"
+    return 1
+  fi
+  file_checksum="$(gzip -c -1 "$3" | tail -c8 | od -t x4 -N 4 -A n | awk '{print $1}' | xxd -r -p | base64)"
+  if ! check_checksum_rest "$1" "$2" "$file_checksum" "x-amz-checksum-crc32"; then
+    log 2 "error checking checksum"
+    return 1
+  fi
+  return 0
+}
+
+crc64_file() {
+  source ./env/bin/activate
+  python3 -m pip install awscrt
+  if ! checksum=$(python3 -c "import sys;from awscrt import checksums;crc64nvme_value = checksums.crc64nvme(sys.argv[1]);print(crc64nvme_value)" "$1"); then
```

Comment:
More of this little built in script - if we are doing repeat code then it really should be stored separately.